[28기 이가윤] 필수/추가 구현사항 완료

Gayun00 opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [x] 리뷰 반영
- [x] 리팩토링
- [x] 버그 수정
- [x] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 피드를 끝까지 스크롤 했을 때 스켈레톤 표시를 거쳐 새로운 피드가 생성되는 기능 구현

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)

**(1) 로그인화면**
1. 로그인 유효성 검사
- disabled 속성을 사용하여 로그인 버튼 활성화/비활성화

<br />
<br />

**(2) 메인화면 - 필수/추가 구현사항**

1. 댓글 추가, 삭제 기능 구현
- document에 이벤트를 등록하여 각 버튼의 클래스와 이벤트 타겟의 클래스네임이 동일할 경우 특정 함수를 실행하도록 코드 작성
- createElement(), appendChild()로 댓글 생성 기능 구현
- removeChild()로 댓글 삭제 기능 구현
- 삭제 버튼 클릭 시 타겟의 클래스이름과 삭제 버튼의 클래스가 동일하면, 해당 노드와 관련된 부모 댓글 노드를 탐색해 삭제

<br />

2. 좋아요 기능 구현
- 좋아요 버튼 클릭 시 liked클래스 추가하여 아이콘 색상 변경
- '--님 외 n명이 좋아합니다' 문자열을 템플릿 리터럴을 사용해 n+1명으로 표시

<br />

3. 계정 검색 기능 구현
- 별도의 계정정보를 담은 배열을 선언하고, filter()와 innerHTML를 사용해 검색한 단어를 포함하는 결과값 표시
- 프로필 이미지와 서브 타이틀 정보가 null 일 경우 표시할 default값 지정

<br />

4. my page 버튼 클릭 시 말풍선 표시 기능 구현
- 말풍선 영역외 다른 영역 클릭 시 말풍선 숨김처리

<br />

5. 반응형 구현
- media query 사용해 화면 창 너비 축소 시 우측 main-right 영역 사라지도록 구현

<br />
<br />

**(3) 메인화면 - 개인적으로 추가한 구현사항**

1. 스크롤 시 새로운 피드 생성
- 각 피드들을 감싸고 있는 하나의 컨테이너 요소 맨 밑에 feedEnd요소 생성
- feedEnd를 intersectionObserver로 관찰하며 화면 안에 들어왔을 때 (맨 밑까지 스크롤 했을 때) 스켈레톤 피드를 새로 생성하는 addSkeletonFeed() 호출
- setTimeout으로 로딩시간 지연 후 displayFeed()호출 > 스켈레톤 대신 feed.json에서 불러온 새로운 피드 정보 표시
- 더 이상 표시할 피드가 없을 경우 '불러올 피드가 없습니다' 메시지 표시

<br />

2. 스토리 모두보기,접기 기능
- 우측 스토리창 모두보기 버튼 클릭시 펼치기, 접기 버튼 클릭시 원 상태로 접기 기능 구현

<br />

3. 스토리 업로드 시간 표시
- 스토리가 업로드된 시간의 데이터를 전달받았다고 가정하고, story.json에 업로드한 시간의 정보를 지정
- 현재 시점에서 스토리별 업로드된 시간부터 경과된 일 수를 계산해 'n일 전' 업로드된 스토리인지 정보 표시

<br />

3. 피드 글 더보기 기능
- 피드의 글이 30자 이상일 경우 글자를 자르고 ...더보기 버튼 표시
- 버튼을 클릭하면 전체 텍스트 표시

<br />
<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
**배운점**
- intersectionObserver에 대해 배웠습니다. debounce, throttle 대신 reflow를 일으키지 않고 요소를 관찰할 수 있는 등의 장점과 사용법에 대해 확실히 알게 되었습니다.
- 검색 결과와 말풍선과 같은 형태를 css로 어떻게 처리할지 모호했었는데, 프로젝트를 하면서 해결했습니다.

 <br />

**아쉬운점**
- css와 html을 체계적으로 작성하지 못한 부분이 있어 아쉽습니다. 예를 들면 css root에서 변수를 --item-padding, --container-padding으로 지정해 사용했는데, 실제 패딩 값이 아닌 간격을 지정하는 데에 빈번하게 쓰였습니다. 너무 많이 쓰여서 일일히 수정할 수는 없어서, 다음 번 프로젝트 때는 해당 변수의 쓰임을 포함한 네이밍에 신경을 더 쓰려고 합니다.
- 로그인 화면을 구현할때 유효성 검사를 구현하는 로직이 깔끔하지 않습니다. 메인화면에 치중해 한번 더 수정하지 못했는데, 정규식을 사용해 깔끔하고 간결한 코드로 다시 만들어보고 싶습니다.

<br />

**어려웠던 점**
- 메인화면 피드 영역 이벤트 처리 방식에 대해 궁금합니다.
이벤트 처리에 있어 특정 버튼을 클릭할 때 변화가 일어나야 하는 다른 요소를 노드(nextSibling, previousSibling)등으로 연결해 접근하고 있습니다. 처음에는 전체document에 이벤트를 준다음 새로 피드가 생성되더라도 한번에 각 요소에 접근해 이벤트를 발생시킬 수 있는 효율적인 방법이 아닐까 싶어서 한번 시도해봤습니다. 그런데 인접한 노드라도 실제로는 접근하기 위해 생각보다 긴 코드가 필요했고, 이후 레이아웃이 변경되거나 코드를 수정해야하는 일이 생긴다면 어떤 노드와 경로를 타고 어떤 것을 지정한 것인지 식별하기 힘들것 같다는 생각이 듭니다. 피드 생성시마다 동일하게 요소별로 이벤트를 따로 따로 등록해주면 될까요? 어떤 더 효율적이고 좋은 방법이 있을지 궁금합니다.
- 위에서 말씀드린 부분 - 노드로 접근하는 방식이 브라우저 성능에 영향을 미치는지 궁금합니다.
이 부분은 검색해도 찾기 어려웠습니다. querySelector로 class 선택자를 사용해 이벤트를 등록하는것보다  추가적인 연산이 필요한가요? 성능과 관련해 css 에서 선택자를 사용할 때 id,class가 아닌 tag와 tag에 대한 자식, 자손 선택자는 성능을 많이 저하시킨다는 내용을 봤는데, 자바스크립트에서도 마찬가지인지 궁금합니다..!

<br />
<br />

bonbon0202 commented on students/15th/kimbyulyi/login.html:
```diff
@@ -0,0 +1,94 @@
+                class="userId"
+              />
+              <input type="password" placeholder="Password" class="userPw" />
+              <button class="loginBtn">Log In</button>
```
class명이 어떤건 BEM, 어떤건 camel case 형식이네요. 통일해주세요! 🙌🏻

kimbyulyi replied:
넵 수정하겠습니다.🙆‍♀️

bonbon0202 commented on students/15th/kimbyulyi/styles/login.css:
```diff
@@ -0,0 +1,179 @@
+}
+
+.login__bar {
+  display: flex;
+  width: 268px;
+  height: 15px;
+  justify-content: center;
+  align-items: center;
```
연관된 속성끼리는 가독성을 위해 인접하게 작성해주세요! 
```suggestion
  display: flex;
  justify-content: center;
  align-items: center;
  width: 268px;
  height: 15px;
```

kimbyulyi replied:
넵 참고하겠습니다!✅

bonbon0202 commented on students/15th/kimbyulyi/styles/login.css:
```diff
@@ -0,0 +1,179 @@
+}
+
+.bar {
+  background-color: var(--color-nobel);
+  width: 100px;
+  height: 1px;
+  padding: 0;
+  margin: 0;
```
css 속성을 적용하는 순서는 아래와 같습니다. 
```
- **css 속성 순서 준수 여부 (레이아웃에 영향을 많이 주는 순서, 인접 속성끼리 묶기)**
    - Layout Properties (position, float, clear, display)
    - Box Model Properties (width, height, margin, padding)
    - Visual Properties (color, background, border, box-shadow)
    - Typography Properties (font-size, font-family, text-align, text-transform)
    - Misc Properties (cursor, overflow, z-index)
```

kimbyulyi replied:
네 참고해서 수정하겠습니다! 감사합니다☺️

bonbon0202 reviewed (commented):
별이님! 코드 깔끔하게 잘 작성해주셨네요!! 🙌🏻
수고 많으셨습니다 :) 
리뷰 반영해서 리펙토링 해주시고, 메인 기능구현도 마저 진행해주세요!

kimbyulyi commented on students/15th/kimbyulyi/js/login.js:
```diff
@@ -0,0 +1,18 @@
+userId.addEventListener("keyup", chnageBtn);
+userPw.addEventListener("keyup", chnageBtn);
+
+// image slide :: 추가 기능
```
넵 확인했습니다! ✅

kimbyulyi commented on students/15th/kimbyulyi/js/login.js:
```diff
@@ -0,0 +1,18 @@
+const loginBtn = document.querySelector(".loginBtn");
+const login__form = document.querySelector(".login__form");
+
+loginBtn.disabled = true;
+
+const chnageBtn = () => {
+  const currentBtn = userId.value && userPw.value;
+  loginBtn.disabled = currentBtn ? false : true;
+};
+
+userId.addEventListener("keyup", chnageBtn);
+userPw.addEventListener("keyup", chnageBtn);
```
네! 순서를 바꿔서 작성했습니다. ✅

kimbyulyi commented on students/15th/kimbyulyi/login.html:
```diff
@@ -0,0 +1,94 @@
+    <script src="js/login.js" defer></script>
+  </head>
+  <body>
+    <!-- 메인-->
```
주석 삭제했습니다!✅

kimbyulyi commented on students/15th/kimbyulyi/login.html:
```diff
@@ -0,0 +1,94 @@
+      <article>
+        <!--aside  왼쪽 이미지-->
+        <aside class="login__img">
+          <img src="img/phone.png" />
```
1. 검색엔진이 페이지에 대한 주제를 잘 해석할수 있게 하여 검색 노출에 도움이 됩니다.
2. 서버에 이상이 생긴 경우 혹은 이미지에 손상된 경우에 해당 이미지가 어떤 이미지인지 alt를 통해 알 수 있다.
수정했습니다 감사합니다.✅

kimbyulyi commented on students/15th/kimbyulyi/login.html:
```diff
@@ -0,0 +1,94 @@
+        <!-- 로그인 섹션 상단 -->
+        <section class="login__container">
+          <div class="login__container__top">
+            <img src="img/logo_text.png" />
```
넵 전체적으로 수정했습니다✅

kimbyulyi commented on students/15th/kimbyulyi/styles/common.css:
```diff
@@ -0,0 +1,67 @@
+
+input:focus {
+  outline: none;
+}
```
수정했습니다 감사합니다✅

kimbyulyi commented on students/15th/kimbyulyi/styles/common.css:
```diff
@@ -0,0 +1,67 @@
+  background-color: var(--color-snow);
+}
+
+/* Global */
```
수정했습니다 감사합니다✅

kimbyulyi commented on students/15th/kimbyulyi/styles/common.css:
```diff
@@ -0,0 +1,67 @@
+  --weight-semi-bold: 600;
+  --weight-regular: 400;
+  --weight-light: 300;
+}
```
감사합니다☺️

kimbyulyi commented on students/15th/kimbyulyi/styles/login.css:
```diff
@@ -0,0 +1,179 @@
+  margin: 0 auto;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  align-items: center;
```
수정했습니다!😭

kimbyulyi commented on students/15th/kimbyulyi/styles/login.css:
```diff
@@ -0,0 +1,179 @@
+/*right section*/
+
+.login__container {
+  display: flex;
+  flex-direction: column;
+  width: 351px;
+  height: 618px;
+  padding: 10px 0px;
+  background-color: var(--color-white);
```
감사합니다☺️

dipsiiiiiiiiii commented on students/15th/kimbyulyi/js/login.js:
```diff
@@ -0,0 +1,35 @@
+  const userIdValue = userId.value;
+  const userPwValue = userPw.value;
+  const errorMsg = document.querySelector(".login__error");
+  if (
+    emailCheck.test(String(userIdValue).toLowerCase()) === false ||
+    userPwValue.length < 5
+  ) {
+    errorMsg.innerText = "계정을 다시 확인해주세요";
+  } else {
+    errorMsg.innerText = "로그인 성공";
+  }
```
if, else 문 말고 삼항 연산자로 이렇게도 해볼 수 있습니다.
```suggestion
  const isValid = emailCheck.test(String(userIdValue).toLowerCase()) && userPwValue.length >= 5
  errorMsg.innerText = isValid ? "로그인 성공" : "계정을 다시 확인해주세요"
```

kimbyulyi replied:
넵! 참고하겠습니다

dipsiiiiiiiiii commented on students/15th/kimbyulyi/js/login.js:
```diff
@@ -0,0 +1,35 @@
+
+const chnageBtn = () => {
+  const currentBtn = userId.value && userPw.value;
+  loginBtn.disabled = currentBtn ? false : true;
```
결과 자체가 boolean 타입이기 때문에 더 줄일 수 있겠네요!
```suggestion
  loginBtn.disabled = !currentBtn
```

kimbyulyi replied:
넵 감사합니다!!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/js/main.js:
```diff
@@ -0,0 +1,72 @@
+const commentList = document.getElementsByClassName("comments")[0];
+const searchInput = document.querySelector(".search__input");
+const searchList = document.querySelector(".search__result");
+let checLikeStatus = false;
```
오타 수정해주세요!

kimbyulyi replied:
넵 checkLikeStatus로 수정하겠습니다!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/js/main.js:
```diff
@@ -0,0 +1,72 @@
+  cmtLikeList.forEach((event) => {
+    event.addEventListener("click", () => {
+      let likeImg = event.querySelectorAll(".like__img")[0];
+      if (checLikeStatus) {
+        likeImg.src = "img/heart.png";
+        checLikeStatus = false;
+      } else {
+        likeImg.src = "img/heart_fill.png";
+        checLikeStatus = true;
+      }
```
다음과 같이 줄일 수 있겠네요!
```suggestion
  likeImg.src = checkLikeStatus ? "img/heart.png" : "img/heart_fill.png";
  checLikeStatus = !checkLikeStatus
```

kimbyulyi replied:
삼항연산자로 코드를 많이 줄일수있네요!! 감사합니다!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/js/main.js:
```diff
@@ -0,0 +1,72 @@
+    if (event.keyCode === 13) postComment(event);
+  });
+
+  likeComment();
+  deleteComment();
```
함수 이름 자체가 댓글 자체를 좋아요 하거나 삭제한다는 함수같이 들리는데
실제로는 event를 연결하는 것이기 때문에 이름을 바꾸시면 더 좋을 것 같습니다!

kimbyulyi replied:
네 멘토님 의견대로 수정하겠습니다!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/styles/common.css:
```diff
@@ -0,0 +1,39 @@
+html,
+body {
+  overflow: auto;
+  width: 100%;
+  height: 100%;
```
이건 초기화라서 reset.css에 들어가야 할 것 같네요!

kimbyulyi replied:
넵!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/login.html:
```diff
@@ -0,0 +1,91 @@
+              <button class="login__btn" type="button">Log In</button>
+            </form>
+            <div class="login__bar">
+              <div class="bar"></div>
+              <span class="bar__text">OR</span>
+              <div class="bar"></div>
```
전부 span으로 하셔도 될 것 같네요!

kimbyulyi replied:
넵 span으로 수정하겠습니다.

dipsiiiiiiiiii commented on students/15th/kimbyulyi/login.html:
```diff
@@ -0,0 +1,91 @@
+            </div>
+          </div>
+          <div class="login__container__bottom">
+            <div class="app__title">
+              <p>
+                Don't have an account?
+                <a href="#" class="signup__btn">Sign up</a>
+              </p>
+            </div>
```
div로 묶을 필요가 없을 것 같아요!
p 태그에 클래스 바로 넣어 줘도 되겠네요!

kimbyulyi replied:
넵!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/login.html:
```diff
@@ -0,0 +1,91 @@
+        </section>
+      </article>
+    </main>
+    <footer>
+      <ul class="footer__list">
+        <li>About</li>
+        <li>Blog</li>
+        <li>Jobs</li>
+        <li>Help</li>
+        <li>Api</li>
+        <li>Privacy</li>
+        <li>Terms</li>
+        <li>Top Accounts</li>
+        <li>Hashtags</li>
+        <li>Locations</li>
+      </ul>
+      <ul class="footer__list">
+        <li>Beauty</li>
+        <li>Dance & Performance</li>
+        <li>Fitness</li>
+        <li>Food & Drink</li>
+        <li>Home & Garden</li>
+        <li>Music</li>
+        <li>Visual Arts</li>
+      </ul>
+      <p>© 2020 Instagram from Facebook</p>
+    </footer>
```
굿!

kimbyulyi replied:
감사합니다!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/main.html:
```diff
@@ -0,0 +1,232 @@
+    <script src="js/main.js" defer></script>
+  </head>
+  <body>
+    <nav>
+      <div class="navbar">
+        <img src="img/logo_text.png" alt="logo" class="logo" />
+        <div class="wrap-input">
+          <input type="text" placeholder="Search" class="search__input" />
+          <img class="search-icon" src="img/search.png" alt="search icon" />
+        </div>
+        <div class="navitem">
+          <img src="img/explore.png" alt="explore icon" />
+          <img src="img/heart.png" alt="heart icon" />
+          <img src="img/user.png" alt="user icon" />
+        </div>
+      </div>
+    </nav>
```
깔끔하네요!
클래스 이름만 camelCase로 해주세요!

kimbyulyi replied:
네! 수정하겠습니다.

dipsiiiiiiiiii commented on students/15th/kimbyulyi/main.html:
```diff
@@ -0,0 +1,232 @@
+        </div>
+      </div>
+    </nav>
+    <main>
+      <div class="main-wrap">
```
main과 main-wrap div 둘 다 필요하나요?

kimbyulyi replied:
멘토님이 올려주신 레이아웃을 보고 나누느라 main 태그를 사용했고 
main-wrap은 왼쪽 오른쪽 div 태그를 일렬로 정렬시키고 두 컨텐츠 넓이를 고정시키느라 사용했습니다! 
다음번엔 좀 더 생각해서 짜 보겠습니다!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/main.html:
```diff
@@ -0,0 +1,232 @@
+              <span>스토리</span>
+              <a>모두 보기</a>
+            </div>
+            <div class="story__recommend">
+              <div class="story__img-wrap">
+                <img src="img/profile_2.jpg" alt="profile" />
+              </div>
+              <div class="profile__name">
+                <a>happydog_a</a>
+                <p>15분전</p>
+              </div>
+            </div>
```
이렇게까지 하나가 반복되네요.
이렇게 같은 레이아웃에 데이터만 바뀔 때는 배열의 map 메소드를 활용할 수 있습니다.

```
const DATA = [{profileImg: "img/profile_2.jpg, userName: "happydog_a", time: "15분전}, ...]

DATA.map(el => {
  return (
    <div class="story__recommend">
              <div class="story__img-wrap">
                <img src={el.profileImg} alt="profile" />
              </div>
              <div class="profile__name">
                <a>{el.userName}</a>
                <p>{el.time}</p>
              </div>
            </div>
  )
})
```

kimbyulyi replied:
감사합니다! 직접 작성해보면서 응용해보겠습니다!

dipsiiiiiiiiii commented on students/15th/kimbyulyi/main.html:
```diff
@@ -0,0 +1,232 @@
+              </div>
+              <div class="feeds__comments">
+                <ul class="comments">
+                  <li>
+                    <a> taterandcollette</a>
+                    <span class="comment__text">Adorable 💙💙💙</span>
+                    <span class="comment__like">
+                      <img
+                        src="img/heart.png"
+                        alt="like icon"
+                        class="like__img"
+                      />
+                    </span>
+                    <span class="comment__delete">
+                      <img
+                        src="img/remove.png"
+                        alt="remove icon"
+                        class="delete"
+                      />
+                    </span>
+                  </li>
```
댓글도 반복되므로 아래 story 리뷰 참고해주세요!

kimbyulyi replied:
네 참고해서 수정하겠습니다! 피드백 감사합니다!

dipsiiiiiiiiii reviewed (commented):
고생하셨습니다!
리뷰 반영해주세요 :)

claire-euni commented:
closed

Gayun00 committed:
[Add] Login 페이지 레이아웃
- 테스트 업로

Gayun00 committed:
[Add] 테스트

Gayun00 committed:
test

Gayun00 committed:
[Add] 코드 업로드
- 기본,추가 구현사항 완료
- css 수정 진행 중

Gayun00 committed:
[Add] 스크롤 끝 도달시 로딩/스켈레톤 구현 진행중
- 더미 데이터 생성
- 스켈레톤 html/css 구현 중

Gayun00 committed:
[Add] 스크롤 끝 도달시 로딩/스켈레톤 구현 진행중
- 스크롤 끝 도달 시 스켈레톤 생성 완료
- 스켈레톤 표시 후 피드 표시 완료
- 단순 html로만 작업, json파일로 동적으로 생성 예정

Gayun00 committed:
[Add] 스크롤 시 로딩 구현 - 댓글 표시 완료

Gayun00 committed:
[Add] 스크롤 시 로딩 후 이미지 로드 - 데이터 없음 표시 완료
- 표시할 데이터가 더 이상 없을 시 메시지 표기
- 피드 이미지 경로 수정

Gayun00 committed:
[Add] 스크롤 시 이미지 로드 - 피드 없음 표시 후 로드 중단
- 표시할 데이터 없을 시 메세지 표기 후 관찰 중단

Gayun00 committed:
[Refactor] 스크롤 시 이미지 로드 - 생성된 피드 이벤트 등록
- 생성된 피드에 addComment, deleteComment, handleLike 함수 이벤트 등록
- 기본 피드 삭제

Gayun00 committed:
[Fix] 스크롤 시 새 피드 생성 - 오류 수정

Gayun00 committed:
css 수정 중: feed, main-right, 변수 추가

Gayun00 committed:
[Modify] css 수정 진행중
- main-right 수정
- css 단위 수정

Gayun00 committed:
[Modify] css 수정중
- 스켈레톤 수정
- my 말풍선, 검색 결과창 수정
- 우측 스토리 창 수정
- 위 스토리 창 추가

Gayun00 committed:
[Modify] css 수정중 - 검색창
- 검색창, my 화살표 추가

Gayun00 committed:
[Modify] css 수정 - main_right.css 추가
- main, main_right, common css 내용 분리

Gayun00 committed:
[Refactor] Html 클래스 추가
- html 리턴 함수 추가

Gayun00 committed:
[Refactor] top story Html 클래스로 추가
- Html 클래스 추가
- story.json 추가

Gayun00 committed:
[Fix] 스크롤 시 이미지 로드기능 수정 중
- 빠르게 스크롤 시 setTimeout으로 지연된 displayFeed()호출과 loading()호출 간의 차이 발생
- 필요 없는 노드가 생성되는 문제 발견
- 시간 지연 후 표시할 데이터 인덱스를 displayFeed()내부에서 증가시키는 것으로 수정, 동일한 이미지 표시되는 문제 수정

Gayun00 committed:
[Fix] 빠르게 스크롤 시 피드 생성 오류 수정완료
- setTimeout 시간 지연으로 인해 unobserve하기 전 빈 요소가 여러개 생성되는 문제 수정

Gayun00 committed:
[Fix] 피드 위에 피드 없음 메세지 표시되는 오류 수정
- 최초 렌더링 시 피드 생성 전 feedEnd observe되어 피드없음 메세지 표시되는 오류
- 이후 observe 중단되어 피드 생성되지 않는 오류
- startObserve() 추가하여 첫 피드 렌더링 후 observe 시작하도록 수정

Gayun00 committed:
[Refactor] main.js 코드 포맷 수

Gayun00 committed:
[Modify] html 시멘틱 태그 수정

Gayun00 committed:
[Add] 우측 스토리 storyData로 채우기, 스토리 펼치고 접기 기능 추가
- story.json에 uploadTime값 추가: 현재시점에서 업로드 후 경과된 일 수 계산해 스토리에 표시
- 모두보기 버튼 클릭시 스토리 펼치기, 버튼은 '접기'로 텍스트 변경
- 접기 버튼 클릭시 스토리 접기

Gayun00 committed:
[Fix] 로그인 화면 유효성 검사 오류 수정
- 글자를 지웠을 때 비밀번호가 5글자 이하이거나 아이디에 @가 포함되어있지 않을 경우 로그인 버튼 비활성화

Gayun00 committed:
[Fix] 스토리 더보기, 댓글 삭제 오류 수정
- 스토리 더보기 창이 접혀있는게 기본값이 되도록 클래스 collapse 추가
- main.js 오타 수정
